packages/calypso-e2e/src/lib/pages/login-page.ts

```diff
@@ -1,35 +1,40 @@
 /**
  * Internal dependencies
  */
-import * as DataHelper from '../../data-helper';
+import { BaseContainer } from '../base-container';
 
 /**
  * Type dependencies
  */
 import { Page } from 'playwright';
 
+const selectors = {
+	loginContainer: '.wp-login__container',
+	username: '#usernameOrEmail',
+	password: '#password',
+	changeAccountButton: '#loginAsAnotherUser',
+};
+
 /**
  * Represents an instance of the calypso Login page.
+ *
+ * @augments {BaseContainer}
  */
-export class LoginPage {
+export class LoginPage extends BaseContainer {
 	/**
 	 * Creates an instance of the Login page.
 	 *
 	 * @param {Page} page Playwright page on which actions are executed.
 	 */
 	constructor( page: Page ) {
-		this.page = page;
-		this.url = DataHelper.getCalypsoURL( 'log-in' );
+		super( page, selectors.loginContainer );
 	}
 
-	page: Page;
-	url: string;
-
-	loginContainerSelector = '.wp-login__container';
-	usernameSelector = '#usernameOrEmail';
-	passwordSelector = '#password';
-	changeAccountSelector = '#loginAsAnotherUser';
-	alreadyLoggedInSelector = '.continue-as-user';
+	async _postInit(): Promise< void > {
+		await this.page.waitForLoadState( 'networkidle' );
+		const container = await this.page.waitForSelector( selectors.loginContainer );
+		await container.waitForElementState( 'stable' );
```

Comment:
Wait to handle the subtle fade-in animation if there is already a logged in session.